src/realm/query_engine.cpp

```diff
@@ -17,9 +17,65 @@
  **************************************************************************/
 
 #include <realm/query_engine.hpp>
+#include <realm/column_table.hpp>
 
 using namespace realm;
 
+namespace realm {
+
+Columns<SubTable>::Columns(const Columns& other, QueryNodeHandoverPatches* patches)
+    : Subexpr2<SubTable>(other)
+    , m_link_map(other.m_link_map, patches)
+    , m_column_ndx(other.m_column_ndx)
+    , m_column(other.m_column)
+{
+    if (m_column && patches)
+        m_column_ndx = m_column->get_column_index();
+}
+
+void Columns<SubTable>::evaluate(size_t index, ValueBase& destination)
```

Comment:
I'd sorta expect this implementation to use `LinkMap` for the actual traversal in the same way that `Columns<Link>` does, since `LinkMap` will need to support traversing subtable relationships anyway in order for ordinary queries (`subtable.column == 1`) to work over subtables. Is there something I'm missing?